src/backend.rs

```diff
@@ -56,7 +44,7 @@ where
 
     pub fn process(&self, u_tx: UnBuiltTransaction<Datum, Redeemer>) -> Result<()> {
         let tx = self.build(u_tx)?;
-        self.txo_record.issue(tx)?;
+        self.txo_record.issue(tx).map_err(|e| Error::TxORecord(e))?;
```

Comment:
If there exists a `From` impl for `TxORecord` -> `NauError`, then the try operator `?` will be enough. You can either write that `From` impl by hand or you can use the `#[from]` attribute:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=50526529a1b9c8c4d009862f19e5e03b

Reply:
That's really cool, didn't know that